### PR TITLE
feat(impl):[TRI-1440] known knows

### PR DESCRIPTION
**Pull request by ds-ext-kmassalski**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The client code for accessing the Digital Twin Registry (central and decentral) is now available as a spring boot maven library. See the README in the irs-registry-client module for more information.
 
 ### Known knowns
-- PLACEHOLDER REMOVE IF EMPTY: risks that were introduced or discovered in the release and are known but not resolved
+- Digital Twin Registry load problem - after few subsequent requests DTR starts answering with 500 Http status
+- ESS Notifications not working with Decentral Digital Twin Registry Service - bpn is not passed
+- Unpredictable 500 Http status returned by EDC during submodel retrievals when full test data is uploaded (EDC version: 0.4.1)
+- High EDC resource demands - each EDC provider postgres uses > 2 CPU-Cores. The database storage gets filled after few days, until then, resource consumption increases. (EDC version: 0.4.1)
```
Comment (ds-jhartmann) [inline CHANGELOG.md:16]: Since EDC 0.4.1 this is not an issue anymore

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ds-jhartmann) [reply CHANGELOG.md:16]: I already closed it

File: CHANGELOG.md
Comment (ds-jkreutzfeld) [inline CHANGELOG.md:1]: Maybe we want to add that we are currently not really managing our resources (e.g. max threads)

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,7 +10,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The client code for accessing the Digital Twin Registry (central and decentral) is now available as a spring boot maven library. See the README in the irs-registry-client module for more information.
 
 ### Known knowns
-- PLACEHOLDER REMOVE IF EMPTY: risks that were introduced or discovered in the release and are known but not resolved
+- Digital Twin Registry load problem - after few subsequent requests DTR starts answering with 500 Http status
+- ESS Notifications not working with Decentral Digital Twin Registry Service - bpn is not passed
+- Unpredictable 500 Http status returned by EDC during submodel retrievals when full test data is uploaded (EDC version: 0.4.1)
+- Lack of rate limiting of requests - IRS allows sending API requests in a massive, automated manner
```
Comment (mkanal) [inline CHANGELOG.md:16]:
```suggestion
- [TRI-1441] - No limiting of requests in paralle - IRS allows sending API requests unlimited
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,7 +10,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Potential denial-of-service (DoS) attack - IRS allows to enter a large number of characters, which are reflected in the response of the server
```
Comment (mkanal) [inline CHANGELOG.md:17]: Could you please more clear which parameter which ticket please?

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,7 +10,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Software related information disclosure - IRS returns redundant information about the type and version of used software
```
Comment (mkanal) [inline CHANGELOG.md:18]: What is this exactly? how to reproduce please add a ticket.

**Review by mkanal — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -10,7 +10,19 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The client code for accessing the Digital Twin Registry (central and decentral) is now available as a spring boot maven library. See the README in the irs-registry-client module for more information.
 
 ### Known knowns
-- PLACEHOLDER REMOVE IF EMPTY: risks that were introduced or discovered in the release and are known but not resolved
+- [TRI-1346] Digital Twin Registry(DTR) - performance issues - after subsequent requests DTR starts responding 500 http status code
+- [TRI-1460] ESS Notifications endpoints are not working in the decentral Digital Twin Registry scenario because endpoints does not provide bpn as a parameter. 
+- [TRI-1455] Random 500 http status code returned by EDC during submodel retrievals. (EDC version: 0.4.1)
```
Comment (ds-jhartmann) [inline CHANGELOG.md:15]: Not the case anymore with the latest IRS and EDC version

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,7 +10,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 ### Known knowns
-- PLACEHOLDER REMOVE IF EMPTY: risks that were introduced or discovered in the release and are known but not resolved
+- [TRI-1460] ESS Notifications endpoints are not working in the decentral Digital Twin Registry scenario because endpoints does not provide bpn as a parameter. 
+- [TRI-1096] No limiting of requests in parallel - IRS allows sending API requests unlimited
+- [TRI-1100] Potential denial-of-service (DoS) attack - IRS allows to enter a large number of characters, which are reflected in the response of the server
+- [TRI-1098] Software related information disclosure - IRS returns redundant information about the type and version of used software
+- [TRI-793] Misconfigured Access-Control-Allow- Origin Header - by intercepting network traffic it could be possible to read and modify any messages that are exchanged with server
+- [TRI-1095] HTTP security headers configuration could be improved and allow for additional protection against some web application attacks
+- Synchronous communication with external services without circuit breaker pattern - potentially could affect IRS resilience when other services becomes non-responsive
```
Comment (mkanal) [inline CHANGELOG.md:19]:
```suggestion
- [TRI-1441] Synchronous communication with shared C-X services without circuit breaker pattern - potentially could affect IRS resilience when other services becomes non-responsive.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -10,7 +10,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Cascading effects of failure when Digital Twin Registry becomes non-responsive - potentially bulkhead pattern could improve IRS resilience
```
Comment (mkanal) [inline CHANGELOG.md:20]:
```suggestion
- [TRI-1441] Cascading effects of failure when Digital Twin Registry becomes non-responsive - potentially bulkhead pattern could improve IRS resilience
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -10,7 +10,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Retry mechanism used inside IRS could potentially affect IRS resilience - DDOS other services on which IRS is dependent, exhaustion of resources and available threads, etc.
```
Comment (mkanal) [inline CHANGELOG.md:21]:
```suggestion
- [TRI-1477] Retry mechanism used inside IRS could potentially affect IRS resilience - DDOS other services on which IRS is dependent, exhaustion of resources and available threads, etc.
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -10,7 +10,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Lack of resources management - max threads, heap limitation, etc.
```
Comment (mkanal) [inline CHANGELOG.md:22]:
```suggestion
- [TRI-1478] Lack of resources management - thread pooling, heap limitation etc.
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -10,7 +10,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- IRS was not scaled to multiple instances, and was not tested in such environment
```
Comment (mkanal) [inline CHANGELOG.md:23]:
```suggestion
- [TRI-1024] IRS does not support scale out on multiple instances
```

**Review by mkanal — CHANGES_REQUESTED** [review]

**Review by mkanal — APPROVED** [review]

**Review by ds-jhartmann — APPROVED** [review]

**Review by ds-jkreutzfeld — APPROVED** [review]

**Pull request merged**